tests/model/test_model.py

```diff
@@ -220,6 +220,70 @@ def test_register_initial_desired_events(self, mocker, initial_data):
             include_subscribers=True,
         )
 
+    @pytest.mark.parametrize(
+        [
+            "to_vary_in_stream_dict",
+            "realm_msg_retention_days",
+            "feature_level",
+            "expect_msg_retention_text",
+        ],
+        [
+            case(
+                {1: {}},
+                None,
+                10,
+                {1: "Indefinite [Organization default]"},
+                id="ZFL=None_no_stream_retention_realm_retention=None",
```

Comment:
Ah, this is the `None` case that I queried elsewhere, where the organization value slips through to the stream level.

Was `None` a realm value, though? I assume it followed the original rule, ie. -1 for indefinite and +n for days, so I'm not sure about None?

The #api-design topic doesn't seem to have clarified this exactly, so perhaps Mateusz could answer in a follow-up or you could refer him here?

Reply:
Right, this is why we were handling the `days == None` case earlier.
To clarify, `None` was a realm value earlier (between server v2-3, as per my testing on the dev server) and it had the same meaning as the current `-1` for indefinite which was changed by @mateuszmandera's [commit](https://github.com/zulip/zulip/commit/7a03e2a7fe259a3410cadbb1a9a4955aa1fae3c7). So to summarize, I believe this was how the server's representation changed over time:
- Server version 2-3: realm value could have `None` for indefinite and +n for days
- Server version 3.0: added stream message retention days, with `None` for org default, `-1` for indefinite and +n for days
- Server version 3+: realm's `None` value was changed to `-1` by Mateusz's above commit.

(I hope Mateusz should get notification from the above mention, still I'll refer him here from the stream to confirm my hypothesis)

Reply:
Ah, that was an interesting dive through old commits :grinning: This overview sounds correct to me based on that. To be a bit more precise, `message_retention_days` was a thing for streams before 3.0 but only in the sense of the backend logic - but it indeed wasn't exposed via the API in any way (the value had to be set manually via `manage.py shell` or db query). I presume that `zulip-terminal` only cares about the API history here though, where indeed this is only a thing since 3.0 - https://github.com/zulip/zulip/commit/c488a35f10ac7b1e5e20fc46302c85ce2364e825